### PR TITLE
#14 Special Shift nodes

### LINKED ISSUE #14: Special Nodes
This issue addresses the content related to special nodes of Shift. Each node should have its subsection with a detailed description, good practices and example use cases. 
These nodes are:

- Python Script Nodes
- Workflow
- Workflow Process 
- Iteration Nodes
  - Iteration
  - IterationEnd
  - ListAccumulator
- Variables
- Condition Equals

Screenshots and new section pages should be added where necessary.

### PR DESCRIPTION
**Pull request by juliavisserinbibo**
## Issue
Closes #14 

## Description 
This PR addresses the content related to special nodes of Shift. 
These nodes are:
- Python Script Nodes
- Workflow Nodes
  - Input & Output
  - Workflow
  - Workflow Process
- Iteration Nodes
   - Iteration
   - IterationEnd
   - ListAccumulator
- Variables
- Condition Equals

### CODE REVIEW

File: docs/reference/nodes/workflow.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+## WorkflowProcess Node
+
+The *WorkflowProcess* node works similarly to the *Workflow* node, however, the execution the referenced workflow will run by an external process. By default, the workflow will be executed in a new instance of the Python, however, the node allows to choose Python interpreter the workflow will run in. 
+
+![WorkflowProcess Node](../../images/nodes/workflow_process.png)
+
+The available python interpreters will depend on the environment variable configuration. To allow the *WorkflowProcess* find new python interpreters, it must be added a new variable to the environment with the following format `SHIFT_PROCESS_<NAME OF THE INTERPRETER>`. The value of this variable must be the path to the desired python interpreter.
+
+>[!EXAMPLE]
+> To add Maya's python interpreter as an option for the *WorkflowProcess* the following should be added to the environment variables:
```
Comment (hectorjimenezinbibo) [inline docs/reference/nodes/workflow.md:27]:
```suggestion
> To add Maya's Python interpreter as an option for the *WorkflowProcess* the following should be added to the environment variables:
```

File: docs/reference/nodes/workflow.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+> 
+>`'SHIFT_PROCESS_MAYA: "<MAYA INSTALLATION FOLDER>/bin/mayapy.exe"'`
+
+### Inputs
+- **File**: This plug of type [FileIn](../nodes/#plugs) is used to reference the workflow file (*.sft* extension) that will be executed by the node. When the `file` plug content is set, the node will automatically update by creating the input and output plugs corresponding to the sourced workflow inputs and outputs.
+- **DCC**: This plug of type [Enum](../nodes/#plugs) determines the Python interpret to be used for the execution of the referenced workflow.
```
Comment (hectorjimenezinbibo) [inline docs/reference/nodes/workflow.md:33]:
```suggestion
- **DCC**: This plug of type [Enum](../nodes/#plugs) determines the Python interpreter to be used for the execution of the referenced workflow.
```

File: docs/reference/nodes/python_script.md [hunk 3]
```diff
@@ -1,12 +1,30 @@
-# Python Script Operator
+# Python Script Node
 
-In order to prototype your workflow or perform specific operations that are not included in Shift's built-in catalogs, you can script your custom operator with Python.  
-The **Python Script** operator offers the possibility to write your own code to tailor the processing of your data the way you want.  
+The *PythonScript* node offers the possibility to write custom Python code to develop new and specific operator behaviors. This feature is useful for prototyping workflows or processing specialized tasks not covered by other Shift nodes.
 
-![Python Script Operator](../images/pyscript_node.png)  
+External Python libraries can be imported to enhance the capabilities of the *PythonScript* node, as long as the enviornment used to launch Shift supports those libraries. To do this, simply import the modules or packages within the *PythonScript* node and use them as you would in any Python code.
```
Comment (hectorjimenezinbibo) [inline docs/reference/nodes/python_script.md:5]:
```suggestion
External Python libraries can be imported to enhance the capabilities of the *PythonScript* node, as long as the environment used to launch Shift supports those libraries. To do this, simply import the modules or packages within the *PythonScript* node and use them as you would in any Python code.
```

File: docs/reference/nodes/conditional.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Conditional Nodes
+
+## ConditionEquals Node
+>[!NOTE]
+> This node is currently a *Beta* operator. Beta operators are supposed to be used for testing purposes only and their behavior may change in the future.
+
+The *ConditionalEquals* node allows to compare if two values are equal. Depending on the result of the comparison, the execution of the workflow will continue through one graph branch or another.
```
Comment (hectorjimenezinbibo) [inline docs/reference/nodes/conditional.md:7]: Maybe here is needed a note to mention avoiding connecting plugs between nodes of the two branches? 
I mean, because for example, one node of one branch cannot be dependent of the output data from a node of the other branch.

File: .vscode/settings.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (hectorjimenezinbibo) [inline .vscode/settings.json:1]: Is this change necessary? It comes from VSCode settings right?

File: docs/reference/nodes/conditional.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (juliavisserinbibo) [reply docs/reference/nodes/conditional.md:7]: Well, that might not be always true. You can have an Object plug type compared to a List plug type for instance. I think that it is responsibility of the developer to understand what can be compared or not.

File: docs/reference/nodes/iterator.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+### Outputs
+- **outValue**: This plug of type [Object](../nodes/#plugs) will output one value of the input iterable object corresponding to the current iteration. This means that the output of this plug will change dynamically for each loop the iteration Node executes. 
+
+## IteratorEnd Node
+The *IteratorEnd* node is used to define the end of a loop started by an *Iterator* node. This means that all nodes between an *Iterator* and a *IteratorEnd* will be part of the execution loop. It is recommended to always have an *IteratorEnd* node per *Iterator* node in a workflow.
+
+![IteratorEnd Node](../../images/nodes/iteratorEnd.png)  
+
+>[!NOTE]
+> It is advised to connect to the *IteratorEnd* node the output generated in the loop via a custom `through` plug.
```
[the remark this answers is not in the document]
Comment (juliavisserinbibo) [reply docs/reference/nodes/iterator.md:26]: That all the data that must be outputted from the loop should go through the IteratorEnd operator. Otherwise, the IteratorEnd node will show up as dirty at the end of the execution. What the node does in it's execution is copying the values from the inputs to the outputs


image
VS
image

File: docs/reference/nodes/iterator.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+## IteratorEnd Node
+The *IteratorEnd* node is used to define the end of a loop started by an *Iterator* node. This means that all nodes between an *Iterator* and a *IteratorEnd* will be part of the execution loop. It is recommended to always have an *IteratorEnd* node per *Iterator* node in a workflow.
+
+![IteratorEnd Node](../../images/nodes/iteratorEnd.png)  
+
+>[!NOTE]
+> It is advised to connect to the *IteratorEnd* node the output generated in the loop via a custom `through` plug. 
+
+## ListAccumulator Node
```
[the remark this answers is not in the document]
Comment (juliavisserinbibo) [reply docs/reference/nodes/iterator.md:28]: List accumulator uses an instance of the list instead of a copy of the list object, and sets the result to the input as well as the output.  Basically, in each iteration, the node updates its list input. 
Given that ListAppend outputs an object copy, I understand that what you had to do is update a global variable with the output of the ListAppend. That wouldn't be needed with the ListAccumulator node.

File: docs/reference/nodes/workflow.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+The *WorkflowProcess* node works similarly to the *Workflow* node, however, the execution the referenced workflow will run by an external process. By default, the workflow will be executed in a new instance of the Python, however, the node allows to choose Python interpreter the workflow will run in. 
+
+![WorkflowProcess Node](../../images/nodes/workflow_process.png)
+
+The available python interpreters will depend on the environment variable configuration. To allow the *WorkflowProcess* find new python interpreters, it must be added a new variable to the environment with the following format `SHIFT_PROCESS_<NAME OF THE INTERPRETER>`. The value of this variable must be the path to the desired python interpreter.
+
+>[!EXAMPLE]
+> To add Maya's python interpreter as an option for the *WorkflowProcess* the following should be added to the environment variables:
+> 
+>`'SHIFT_PROCESS_MAYA: "<MAYA INSTALLATION FOLDER>/bin/mayapy.exe"'`
```
[the remark this answers is not in the document]
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md:29]: I was thinking of doing it in each instalation page.

File: docs/reference/nodes/conditional.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (JorgeHernandezibanezinbibo) [reply docs/reference/nodes/conditional.md:7]: Ok you are right, let´s skip this. Thx

File: docs/reference/nodes/iterator.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (JorgeHernandezibanezinbibo) [reply docs/reference/nodes/iterator.md:26]: Wow, i didn´t know that this was supported or designed in this way.
Then I have a question, whar value does the IteratorEnds returns? the value from the last iteration?

File: docs/reference/nodes/iterator.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (JorgeHernandezibanezinbibo) [reply docs/reference/nodes/iterator.md:28]: Ok, so the difference would be this one?
image
I'm trying to understand it to know if the documentation it's ok, because I see it a littlebit difficult to imagine how to use.

File: docs/reference/nodes/workflow.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (JorgeHernandezibanezinbibo) [reply docs/reference/nodes/workflow.md:31]: ok then, we can let this like an example only.

File: docs/reference/nodes/iterator.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (juliavisserinbibo) [reply docs/reference/nodes/iterator.md:28]: You wouldn't necessarily need to connect the list variable to the list accumulator. The default value of the inList plug is already a list.

File: .vscode/settings.json (hunk 5 above)
Comment (juliavisserinbibo) [reply .vscode/settings.json:1]: No, it's just my personal configuration for a spell checker. I accidentally pushed it

File: docs/reference/nodes/conditional.md (hunk 4 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/conditional.md:7]: Can you attach an example workflow of what setup you refer to? I think I get it, but just to make it clear

File: docs/reference/nodes/python_script.md [hunk 9]
```diff
@@ -1,12 +1,30 @@
...
+- **Script**: This plug of type [Code](../nodes/#plugs) defines the logic of the *PythonScript* node. On the [Inspector](../../getting_started/basics/ui_overview.md/#inspector), this plug provides a script editor for Python to write the block of code to be executed.
+
+The *PythonScript* node allows for any type of custom plugs too. Custom input plugs can receive data from other nodes that can be imported into the code by just using the plug name as a variable.
+
+
+### Outputs
+
+- **Output**: This plug of type [Object](../nodes/#plugs) outputs the result of the logic executed by the *PythonScript*. To set the result of to this plug, simply equal to `output` (the plug's name) the value to output.
+
+If outputting several values is needed, new outputs plugs can be added to the node, which can be used the same way as the default output plug.
```
[the remark this answers is not in the document]
Comment (juliavisserinbibo) [reply docs/reference/nodes/python_script.md]: Rewrote the sentence

File: .vscode/settings.json (hunk 5 above)
Comment (juliavisserinbibo) [reply .vscode/settings.json]: I removed it

File: docs/reference/nodes/conditional.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+### Outputs
+- **true**: This plug of type [Bool](../nodes/#plugs) indicates if the compared values are equal. If that is the case (the value of this plug is `True`), the nodes connected to this plug will be executed.
+- **false**: This plug of type [Bool](../nodes/#plugs) indicates if the compared values are not equal. If that is the case (the value of this plug is `True`), the nodes connected to this plug will be executed.
+
+<!-- ### Examples
+
+This section is reserved to an example video of how to use the Conditional nodes.
+
+ -->
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/conditional.md:23]: We should add a section here explaining the details of connecting the true and false plugs to the triggers of the nodes that should be executed.

We can also add an example worflow in shift to point to or a screenshot to a very basic workflow execution result.

File: docs/reference/nodes/iterator.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+
+>[!NOTE]
+> *Iterator* Nodes can be nested, meaning that another *Iterator* node can be placed inside the loop of an *Iterator* node to achieve multiple levels of iteration.
+
+### Inputs
+
+- **inValues** : This plug of type [Object](../nodes/#plugs) expects an iterable Python object, such as lists, dictionaries, sets or tuples. The number of elements of the iterable object will determine the number of loops the *Iterator* node will execute.
+
+### Outputs
+- **outValue**: This plug of type [Object](../nodes/#plugs) will output one value of the input iterable object corresponding to the current iteration. This means that the output of this plug will change dynamically for each loop the iteration Node executes.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/iterator.md:18]:
```suggestion
### Outputs

- **outValue**: This plug of type [Object](../nodes/#plugs) will output one value of the input iterable object corresponding to the current iteration. This means that the output of this plug will change dynamically for each loop the iteration Node executes. 

```

File: docs/reference/nodes/iterator.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
+# Iteration Nodes
+
+Iteration Nodes are used to create and define loops within a workflow. 
+
+## Iterator Node
+The *Iterator* node is used to create a loop in a workflow. This node generates multiple execution subgraphs with the nodes following the *Iterator* node. Each execution of the loop uses one value from the input iterable object as if it was effectively iterating over its elements.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/iterator.md:6]:
```suggestion
## Iterator Node

This node forces the execution of the subgraph composed by the nodes following the *Iterator* node as many times as the number of elements passed to the input of the node. Each execution of the loop uses one value from the input iterable object as if it was effectively iterating over its elements.

```

File: docs/reference/nodes/iterator.md [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+### Inputs
+
+- **inValues** : This plug of type [Object](../nodes/#plugs) expects an iterable Python object, such as lists, dictionaries, sets or tuples. The number of elements of the iterable object will determine the number of loops the *Iterator* node will execute.
+
+### Outputs
+- **outValue**: This plug of type [Object](../nodes/#plugs) will output one value of the input iterable object corresponding to the current iteration. This means that the output of this plug will change dynamically for each loop the iteration Node executes. 
+
+## IteratorEnd Node
+The *IteratorEnd* node is used to define the end of a loop started by an *Iterator* node. This means that all nodes between an *Iterator* and a *IteratorEnd* will be part of the execution loop. It is recommended to always have an *IteratorEnd* node per *Iterator* node in a workflow.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/iterator.md:21]:
```suggestion
The *IteratorEnd* node is used to define the end of a loop started by an *Iterator* node. This means that only the nodes between the *Iterator* and the *IteratorEnd* will compose the subgraph used for the execution loop. It is recommended to always have one *IteratorEnd* node per *Iterator* node in a workflow.

```

File: docs/reference/nodes/iterator.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (albertochimentiinbibo) [reply docs/reference/nodes/iterator.md:26]: Ideally you would accumulate the results of each iteration in whatever you pass through the iteratorEnd node so yes it will contain the value set at the last iteration.

File: docs/reference/nodes/iterator.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (albertochimentiinbibo) [reply docs/reference/nodes/iterator.md:28]: Just to add, I belive ListAccumulator nodes were developed a lot before workflow variables so what you are doing here with Variable nodes was possible before only using ListAccumulators.

File: docs/reference/nodes/python_script.md [hunk 14]
```diff
@@ -1,12 +1,30 @@
-# Python Script Operator
+# Python Script Node
 
-In order to prototype your workflow or perform specific operations that are not included in Shift's built-in catalogs, you can script your custom operator with Python.  
-The **Python Script** operator offers the possibility to write your own code to tailor the processing of your data the way you want.  
+The *PythonScript* node offers the possibility to write custom Python code to develop new and specific operator behaviors. This feature is useful for prototyping workflows or processing specialized tasks not covered by other Shift nodes.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/python_script.md:3]:
```suggestion
The *PythonScript* node offers the possibility to write custom Python code to develop new and specific execution behaviors. This feature is useful for prototyping workflows or processing specialized tasks not covered by other Shift nodes.
```

File: docs/reference/nodes/python_script.md [hunk 15]
```diff
@@ -1,12 +1,30 @@
...
-You can also import external Python libraries to extend the functionality of your custom operator as long as the Python interpreter you used to start Shift has them available. To do that simply import the modules or packages inside the Python Script operator and use them as you would in any Python code.  
+### Inputs
 
-This makes Shift compatible with any software that supports a Python API.
+- **Script**: This plug of type [Code](../nodes/#plugs) defines the logic of the *PythonScript* node. On the [Inspector](../../getting_started/basics/ui_overview.md/#inspector), this plug provides a script editor for Python to write the block of code to be executed.
+
+The *PythonScript* node allows for any type of custom plugs too. Custom input plugs can receive data from other nodes that can be imported into the code by just using the plug name as a variable.
+
+
+### Outputs
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/python_script.md:16]:
```suggestion
The *PythonScript* node allows for any type of custom plugs too. Custom input plugs can receive data from other nodes that can be imported into the code by just using the plug name as a variable.

### Outputs
```

File: docs/reference/nodes/python_script.md [hunk 16]
```diff
@@ -1,12 +1,30 @@
...
+
+>[!NOTE]
+> Plugs in *PythonScript* nodes cannot be renamed, as it is important to keep the plug's name unique and consistent for the node's correct function. 
+
+
+<!-- ### Examples
+
+This section is reserved to an example video of how to use the Python Script node.
+
+ -->
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/python_script.md:30]: Can we add an example gif showing the custom input and output plugs functionality

File: docs/reference/nodes/variable.md [hunk 17]
```diff
@@ -0,0 +1,22 @@
+# Variable Node
+
+The *Variable* node provides a way to save and reuse data throughout the workflow. Plugs set in a *Variable* node will be treated as global variables of the workflow, allowing their values to be accessed at any point in the graph. This node is helpful when multiple nodes across large workflows need to share data.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/variable.md:3]:
```suggestion
The *Variable* node provides a way to save and reuse data throughout the workflow. Plugs set in a *Variable* node will be treated as global variables of the workflow, allowing their values to be accessed at any point in the graph. This node is helpful when multiple nodes across large workflows need to share data or a certain variable needs to be updated rather then set for each step of an iteration loop.

```

File: docs/reference/nodes/variable.md [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+
+![Variable Node](../../images/nodes/variable.png)  
+
+### Inputs
+
+When an input plug is added in a *Variable* node, Shift will create an internal workflow global variable with the name of the plug. When the *Variable* node gets executed, the global variable will be updated with the value set in the input plug of the node. 
+
+### Outputs
+
+Output plugs in the *Variable* node are used to retrieve the data from the global variables with a **matching plug name**. When the *Variable* node gets executed, the global variable value will be set as the value of the output plug.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/variable.md:13]:
```suggestion
Output plugs in the *Variable* node are used to retrieve the data from the global variables with a **matching plug name**. When the *Variable* node gets executed, the global variable value will be set as the value of the output plug with matching name.

```

File: docs/reference/nodes/variable.md [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+
+>[!NOTE]
+> Plugs in *Variable* nodes cannot be renamed, as it is important to keep the plug's name unique and consistent for the node's correct function. 
+
+<!-- ### Examples
+
+This section is reserved to an example video of how to use the Variable node.
+
+ -->
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/variable.md:22]: We definitely need a couple of examples of use for the variable nodes here. One very important one is to show how to update for example a dictionary in an iteration loop. Or pass a different values to the same plug depending on the condition equals executed branch.

File: docs/reference/nodes/workflow.md [hunk 20]
```diff
@@ -0,0 +1,44 @@
+# Workflow Nodes
+
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/workflow.md:2]: Can we add a brief text here to explain context of what workflow nodes are used for?

File: docs/reference/nodes/workflow.md [hunk 21]
```diff
@@ -0,0 +1,44 @@
+# Workflow Nodes
+
+
+## Input & Output Nodes
+*Input* and *Output* nodes are used to define the required inputs and outputs of a workflow to be executed externally. Setting these nodes is essential to be able to nest workflows or executing the workflow in [batch mode](../../getting_started/basics/batch.md). The custom plugs from these nodes, which should be of type `through`, will act as data pathways, linking information set externally in the workflow to the current workflow and viceversa.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/workflow.md:5]:
```suggestion
*Input* and *Output* nodes are used to define the required inputs and outputs of a workflow to be executed externally. Setting these nodes is essential to be able to nest workflows or executing the workflow in [batch mode](../../getting_started/basics/batch.md). The custom plugs from these nodes, which should be of type `through`, will act as data pathways, linking information set externally in the workflow to the current workflow and viceversa. These nodes will also be crucial to pass external inputs or retreiving results when executing Shift workflows in Batch mode.

```

File: docs/reference/nodes/workflow.md [hunk 22]
```diff
@@ -0,0 +1,44 @@
...
+
+
+## Input & Output Nodes
+*Input* and *Output* nodes are used to define the required inputs and outputs of a workflow to be executed externally. Setting these nodes is essential to be able to nest workflows or executing the workflow in [batch mode](../../getting_started/basics/batch.md). The custom plugs from these nodes, which should be of type `through`, will act as data pathways, linking information set externally in the workflow to the current workflow and viceversa. 
+
+
+![Workflow Node](../../images/nodes/Input_outputNodes.png) 
+
+## Workflow Node
+The *Workflow* node executes an external workflow different from the workflow the node is located in, allowing, this way, to nest workflows. The node expands its plugs dynamically to enable access to the referenced workflow input and output data.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/workflow.md:11]:
```suggestion
The *Workflow* node executes an external workflow. This node allows to nest workflows embedding the execution logic of entire Shift workflows in one single node. The node updates its plugs dynamically depending on the sourced workflow file to enable access to the referenced workflow input and output data.

```

File: docs/reference/nodes/workflow.md [hunk 23]
```diff
@@ -0,0 +1,44 @@
...
+### Inputs
+
+- **File**: This plug of type [FileIn](../nodes/#plugs) is used to reference the workflow file (*.sft* extension) that will be executed by the node. When the `file` plug content is set, the node will automatically update by creating the input and output plugs corresponding to the sourced workflow inputs and outputs.
+
+## WorkflowProcess Node
+
+The *WorkflowProcess* node works similarly to the *Workflow* node, however, the execution of the referenced workflow will be run by an external process. By default, the workflow will be executed in a new instance of Python, however, the node allows to choose Python interpreter the workflow will run in. 
+
+
+![WorkflowProcess Node](../../images/nodes/workflow_process.png)
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/workflow.md:24]:
```suggestion
The *WorkflowProcess* node works similarly to the *Workflow* node, however the execution of the referenced workflow will be handled by an external process. By default, the workflow will be executed in a new instance of the default Python interpreter but the node allows to source and choose different ones. 

![WorkflowProcess Node](../../images/nodes/workflow_process.png)

```

File: docs/reference/nodes/workflow.md [hunk 24]
```diff
@@ -0,0 +1,44 @@
...
+
+The available Python interpreters will depend on the environment variable configuration. To allow the *WorkflowProcess* find new Python interpreters, it must be added a new variable to the environment with the following format `SHIFT_PROCESS_<NAME OF THE INTERPRETER>`. The value of this variable must be the path to the desired Python interpreter.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/workflow.md:26]:
```suggestion
The available Python interpreters will depend on the environment configuration. To allow the *WorkflowProcess* to find new Python interpreters, a new environment variable must be set with the following format `SHIFT_PROCESS_<NAME OF THE INTERPRETER>`. The value of this variable must be the path to the desired Python interpreter (.exe file for Windows users).

```

File: docs/reference/nodes/workflow.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (albertochimentiinbibo) [reply docs/reference/nodes/workflow.md:31]: can we change the formatting of the example? Using "variable": "value" is no legit way of setting a variable in any OS. Can we do:
```suggestion
> In Windows for example:
>`'set SHIFT_PROCESS_MAYA=<MAYA INSTALLATION FOLDER>/bin/mayapy.exe'`

```
or something similar?

**Review by albertochimentiinbibo — CHANGES_REQUESTED** [review]

File: docs/reference/nodes/iterator.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (JorgeHernandezibanezinbibo) [reply docs/reference/nodes/iterator.md:26]: Thx for the explanation. Then, I would suggest to specify this in the Note. I think it's usefull to clarify that if you connect plugs in the IteratorEnd and you user the outputs latter, the value get outside the loop will be the value from the last iteration. What could be used for the ListAccumulator, if we want to add that.

File: docs/reference/nodes/iterator.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (JorgeHernandezibanezinbibo) [reply docs/reference/nodes/iterator.md:28]: Ok I see. Before you could not save values, I sse the needed of this before the variables. nothing to modify then. Thx!

File: docs/reference/nodes/conditional.md (hunk 10 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/conditional.md:23]: You would add it in the examples section or in the general explanation? What kind of details would you include about that topic?

File: docs/reference/nodes/variable.md (hunk 18 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/variable.md:13]: We already mention that it must have a matching name in the previous sentence

File: docs/reference/nodes/workflow.md (hunk 20 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md:2]: Yes! I completely missed that, thanks

File: docs/reference/nodes/conditional.md (hunk 4 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/conditional.md:7]: Hmm I understand that this may not be a desired logic in many cases, but it does not break anything, right? I mean, it is like using a variable in a code that gets changed by an if statement that you use later in the code. 
Do you think is worth adding a recommendation node about this anyways?

File: docs/reference/nodes/workflow.md (hunk 21 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md:5]: The paragraph already mentions this it is used for batch execution. Should we be more specific for what having this nodes means for interactive mode vs batch mode?

File: docs/reference/nodes/variable.md (hunk 17 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/variable.md]: Done

File: docs/reference/nodes/workflow.md (hunk 22 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md]: Done

File: docs/reference/nodes/workflow.md (hunk 24 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md]: Almost done. I'm waiting to know if we should make specific mentions to windows or not.

File: docs/reference/nodes/conditional.md (hunk 4 above)
Comment (hectorjimenezinbibo) [reply docs/reference/nodes/conditional.md:7]: Well, if it doesn't break anything we can leave it, when the users test that case they will realise that the other branch won't be executed. 👍

File: docs/reference/nodes/python_script.md [hunk 25]
```diff
@@ -1,12 +1,31 @@
...
-This makes Shift compatible with any software that supports a Python API.
+- **Script**: This plug of type [Code](../nodes/#plugs) defines the logic of the *PythonScript* node. On the [Inspector](../../getting_started/basics/ui_overview.md/#inspector), this plug provides a script editor for Python to write the block of code to be executed.
+
+The *PythonScript* node allows for any type of custom plugs too. Custom input plugs can receive data from other nodes that can be imported into the code by just using the plug name as a variable.
+
+### Outputs
+
+- **Output**: This plug of type [Object](../nodes/#plugs) outputs the result of the logic executed by the *PythonScript*. To set the result of to this plug, simply equal to `output` (the plug's name) the value to output.
+
+If outputting several values is needed, new output plugs of any type, including Instance plugs, can be added to the node. All custom outputs can be used the same way as the default output plug to return any data from the python script execution.
```
Comment (hectorjimenezinbibo) [inline docs/reference/nodes/python_script.md:19]:
```suggestion
If outputting several values is needed, new output plugs of any type, including Instance plugs, can be added to the node. All custom outputs can be used the same way as the default output plug to return any data from the Python script execution.
```

**Review by hectorjimenezinbibo — CHANGES_REQUESTED** [review]

**Review by JorgeHernandezibanezinbibo — APPROVED** [review]

**Review by hectorjimenezinbibo — APPROVED** [review]

File: docs/reference/nodes/workflow.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (albertochimentiinbibo) [reply docs/reference/nodes/workflow.md:30]: yes, whatever decision we take there it is ok to follow the same format for me.

File: docs/reference/nodes/conditional.md (hunk 10 above)
Comment (albertochimentiinbibo) [reply docs/reference/nodes/conditional.md:23]: the conditionEquals node has a specific logic of connecting the true and false plugs to the triggers of the nodes that should be executed. This is a very special behaviour that should be explained. I would add a section to explain how to use the node.

File: docs/reference/nodes/python_script.md (hunk 16 above)
Comment (albertochimentiinbibo) [reply docs/reference/nodes/python_script.md:31]: thank you

File: docs/reference/nodes/variable.md (hunk 18 above)
Comment (albertochimentiinbibo) [reply docs/reference/nodes/variable.md:13]: not very clear within the previous section ayways what we talk about before is setting the workflow vaiable with a name matching the input plug name. Here we say that the output plug is set to the value of the matching workflow variable. It is slightly different.

We can skip adding this specification if you do not like it. In my opinion it does no harm to repeat important concepts.

Feel free to resolve with the solution you prefer.

File: docs/reference/nodes/variable.md (hunk 19 above)
Comment (albertochimentiinbibo) [reply docs/reference/nodes/variable.md:22]: ok makes sense. Let's add a TODO as I think the documentation will not be complete if we do not add a gif or a video.

File: docs/reference/nodes/workflow.md (hunk 21 above)
Comment (albertochimentiinbibo) [reply docs/reference/nodes/workflow.md:5]: my bad!

File: docs/reference/nodes/workflow.md (hunk 23 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md]: I don't think so now that you mention it. The list of available python processes depends on that variable. If not set, no interpreter will be set and the node will error.

File: docs/reference/nodes/workflow.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md:30]: So, we change the format to use bold instead of code quotes

[the remark this answers is not in the document]
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md]: Changed to bold enviorment variable and italic path

File: docs/reference/nodes/conditional.md (hunk 10 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/conditional.md:32]: I added an image right below

File: docs/reference/nodes/workflow.md (hunk 24 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md]: It is specified in the example for maya, not the general description

File: docs/reference/nodes/workflow.md (hunk 23 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md]: I added more extended explanation and a link to the installation page

**Review by albertochimentiinbibo — APPROVED** [review]
Approving, please reply to the pending comments.

File: docs/reference/nodes/workflow.md [hunk 26]
```diff
@@ -0,0 +1,51 @@
...
+The *Workflow* node executes an external workflow. This node allows to nest workflows embedding the execution logic of entire Shift workflows in one single node. The node updates its plugs dynamically depending on the sourced workflow file to enable access to the referenced workflow input and output data.
+
+<figure markdown>
+    <img src="images/workflow.png" alt="Workflow Node">
+    <figcaption><b>Figure 2</b>: Workflow Node UI.</figcaption>
+</figure>
+
+### Inputs
+
+- **File**: This plug of type [FileIn](../nodes#plugs) is used to reference the workflow file (*.sft* extension) that will be executed by the node. When the `file` plug content is set, the node will automatically update by creating the input and output plugs corresponding to the sourced workflow inputs and outputs.
```
Comment (albertochimentiinbibo) [inline docs/reference/nodes/workflow.md:22]: Please add a section to explain the double click functionality

**Review by albertochimentiinbibo — CHANGES_REQUESTED** [review]

File: docs/reference/nodes/workflow.md (hunk 26 above)
Comment (juliavisserinbibo) [reply docs/reference/nodes/workflow.md]: Done

File: docs/reference/nodes/workflow.md [hunk 27]
```diff
@@ -0,0 +1,56 @@
...
+## WorkflowProcess Node
+
+The *WorkflowProcess* node works similarly to the *Workflow* node, however, the execution of the referenced workflow will be handled by an external process running in the background.
+
+<figure style="width:80%;">
+    <img src="images/workflow_process.png" alt="WorkflowProcess Node">
+    <figcaption><b>Figure 3</b>: WorkflowProcess Node UI.</figcaption>
+</figure>
+
+As well as the *Workflow* node, by double-clicking on the *WorkflowProcess* node, a new tab with the referenced workflow will be opened.
```
Comment (juliavisserinbibo) [inline docs/reference/nodes/workflow.md:36]:
```suggestion
As well as the *Workflow* node, by double-clicking on the *WorkflowProcess* node, a new board with the referenced workflow will be opened.

```

File: docs/reference/nodes/workflow.md [hunk 28]
```diff
@@ -0,0 +1,56 @@
...
+## Workflow Node
+The *Workflow* node executes an external workflow. This node allows to nest workflows embedding the execution logic of entire Shift workflows in one single node. The node updates its plugs dynamically depending on the sourced workflow file to enable access to the referenced workflow input and output data.
+
+<figure>
+    <img src="images/workflow.png" alt="Workflow Node">
+    <figcaption><b>Figure 2</b>: Workflow Node UI.</figcaption>
+</figure>
+
+
+When double-clicking on a *Workflow* node, the referenced workflow will be opened in a new tab inside Shift. Any changes done to the referenced workflow will be picked up by the *Workflow* node and update its plugs if necessary.
```
Comment (juliavisserinbibo) [inline docs/reference/nodes/workflow.md:21]:
```suggestion
When double-clicking on a *Workflow* node, the referenced workflow will be opened in a new board inside Shift. Any saved changes done to the referenced workflow will be picked up by the *Workflow* node and its plugs will be updated if necessary.

```

**Review by albertochimentiinbibo — APPROVED** [review]
